system-addon/content-src/components/SectionMenu/SectionMenu.jsx

```diff
@@ -0,0 +1,53 @@
+import {actionCreators as ac} from "common/Actions.jsm";
+import {ContextMenu} from "content-src/components/ContextMenu/ContextMenu";
+import {injectIntl} from "react-intl";
+import React from "react";
+import {SectionMenuOptions} from "content-src/lib/section-menu-options";
+
+const DEFAULT_SECTION_MENU_OPTIONS = ["RemoveSection", "CheckCollapsed", "Separator", "ManageSection"];
+
+export class _SectionMenu extends React.PureComponent {
+  getOptions() {
+    const {props} = this;
+
+    const propOptions = props.options || DEFAULT_SECTION_MENU_OPTIONS;
+
+    const options = propOptions.map(o => SectionMenuOptions[o](props)).map(option => {
```

Comment:
This would seem like it could be shared with `LinkMenu.jsx`'s functionality as higher order function probably from a shared `content-src/lib/menu-options`, e.g., `makeOptions(propOptions, SectionMenuOptions)`? @sarracini 

Although now I realize React is probably rerendering this component like crazy because there's all these dynamic function bindings for `onClick` even though it probably could just be defined once and refer to `this` to get at `options`?

I guess all of this HOF and more cleanup should be a separate bug. Would be interesting to see if this is actually has performance impacts given how many context menus we have now… ~25 menus * ~10 options -> 250 `ContextMenuItem` being rendered every time a prop updates?

Reply:
Yeah this little chunk here could be refactored to a more general "menu options" instead of a link menu and a sections menu. But I think you're right @Mardak, we should evaluate the performance and do it as a follow up

Reply:
One thing we can easily do is instead of having:

```
  render() {
    return (<ContextMenu
      visible={this.props.visible}
      onUpdate={this.props.onUpdate}
      options={this.getOptions()} />);
  }
```

we should have:

```
  render() {
    return this.props.visible && (<ContextMenu
      visible={this.props.visible}
      onUpdate={this.props.onUpdate}
      options={this.getOptions()} />);
  }
```

That we we render on demand. Should save all the cpu cycles and kittens

Reply:
I think the "react way" is something like this anyway:

```
{this.props.visible &&
  <Component />
}
```

Reply:
Yeah. I played with it but it requires changes in `<ContextMenu/>` because it currently assumes that it will be mounted initially with `visible=false`. But should be easyish fix. Just need to do `LinkMenu` and `SectionMenu` changes together, so I'd rather do as followup instead of piling on unrelated `LinkMenu` changes here.

Reply:
Filed https://bugzilla.mozilla.org/show_bug.cgi?id=1440456